client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -1,140 +1,51 @@
-/**
- * External dependencies
- */
-
-import EventEmitter from 'events';
-import { defer, uniqueId } from 'lodash';
 /**
  * Internal dependencies
  */
-import EmbedsListStore from 'lib/embeds/list-store';
-import EmbedsStore from 'lib/embeds/store';
-import actions from 'lib/embeds/actions';
 import EmbedView from './view';
+import getSiteEmbeds from 'state/selectors/get-site-embeds';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { SELECTED_SITE_SUBSCRIBE, SELECTED_SITE_UNSUBSCRIBE } from 'state/action-types';
+import { reduxDispatch, reduxGetState } from 'lib/redux-bridge';
+import { requestEmbed, requestEmbeds } from 'state/embeds/actions';
 
-export default class EmbedViewManager extends EventEmitter {
-	constructor() {
-		super();
+const EmbedViewManager = {};
 
-		this.sitesListener = this.updateSite.bind( this );
+EmbedViewManager.match = content => {
```

Comment:
This new implementation of `match` has two flaws:

First, it dispatches `requestEmbeds` on each call, even if the embed list is already loaded. The original implementation didn't do that. That means that the `/embeds` REST request is sent very frequently as the user edits the content.

Second, if on the first call to `match` the embeds list is not yet loaded, it doesn't find any of the embeds. And when the list load is finished, we need to somehow trigger the `match` again. The original implementation achieved that by being an `EventEmitter` instance and becoming part of the [`emitters` list](https://github.com/Automattic/wp-calypso/blob/11374ce9c2da867f791832327cdcbb85f0b6537e/client/components/tinymce/plugins/wpcom-view/views.js#L36). The new implementation doesn't subscribe to changes in the Redux store at all.

Reply:
Thanks, really good catch. 

I was actually able to repro the issue you're talking about (the second one) by delaying the initial embeds request to take longer. Unfortunately, it seems like the TinyMCE plugins rely on the emitters quite a bit. This is something we can't really get away from in this PR. It's likely a worthy effort, but not for this PR.

So, I've reverted the `EmbedViewManager` to an `EventEmitter`, but changed it so it now subscribes to changes of interest in the Redux store. That way, `match` will be called again whenever the embed patterns are loaded, and the embeds will be properly rendered.

Furthermore, I've modified it so we'll only request site embed patterns once. 

Let me know what you think.